### ISSUE TITLE
Parameters not working in embedded reports "Assert failed: (map? token-params)"

### ISSUE THREAD

**Issue by adjenks** [metabase/metabase#11101]
**Describe the bug**
Embedded reports return 400: Assert failed: (map? token-params)

**Logs**
Javascript console:
`Error getting setting scalar.field TypeError: "u(...)[0].data is undefined"`
Network:
- Path: `/api/embed/dashboard/<key>/dashcard/25/card/27?testvariable=foobar`
- Return code: `400`
- Return body:`Assert failed: (map? token-params)`

**To Reproduce**
1. Create a question `SELECT {{testvariable}}`
2. Create a dashboard with that question and a filter of the same name
3. Embed, with the parameter set to editable.

**Expected behavior**

**Screenshots**
Works fine in the preview:
![image](https://user-images.githubusercontent.com/10687569/66444181-09ac6e00-e9f7-11e9-8769-9b8e9ea59cf9.png)

Doesn't work embedded:
![image](https://user-images.githubusercontent.com/10687569/66444159-f5687100-e9f6-11e9-87b8-4e466af6f2e5.png)

**Information about your Metabase Installation:**
- Your browser and the version: Firefox and Chrome
- Your operating system: CentOS 
- Your databases: Postgres
- Metabase version: v0.33.0
- Metabase hosting environment: CentOS
- Metabase internal database: Postgres

**Severity**
Severe, can't embed dashboards.

**Additional context**
The parameters work fine when using a public/insecure iframe url.

**Comment by flamber** [comment]
Hi @adjenks 
I cannot reproduce your issue on 0.33.4.
It sounds like you're not passing the `params` - you need to pass it, even when it's empty.
Which language are you using for the embedding? Can you post the code?

**Comment by adjenks** [comment]
@salsakran 
I'm using php, I include the empty params, if I do not include them it gives a different error than the one I described. Without adding empty params, it says "Token is missing value for keypath". This is not the problem, I do add the empty params, as shown below.

```php
		$signer = new Sha256();
		$token = (new Builder())
			->set('resource', [
				'dashboard' => 5
			])
			->set('params', [

			])
			->sign($signer, $metabaseSecretKey)
			->getToken();
		$iframeUrl = "$metabaseSiteUrl/embed/dashboard/$token";
		$response->getBody()->write(
			"<iframe src=\"$iframeUrl\"
			width=\"1000\"
			height=\"800\"
			frameborder=\"0\" />"
		);
```

**Comment by adjenks** [comment]
@salsakran 

I figured it out. It has to do with how PHP handles arrays. In PHP
- `[]`: An empty Array, defaults to numeric and serializes to JSON: `[]`
- `['a':1]`: Still an array, but an associative array serializes to JSON: `{"a":1}`
- `(object)[]`: A PHP array cast to an object, serialized to JSON: `{}`

By casting to an object, I've solved the issue.
It would be nice if the examples provided in your example repo would be updated with this detail. They currently only show example that have a single parameter being passed, not zero.

I figured it out after reading some of the source and seeing the validation line:
https://github.com/metabase/metabase/blob/891e128b1f3dfad7e73250e54108148cba491678/src/metabase/api/embed.clj#L217
I don't know clojure so I had to read a bit to understand what `map?` does. It was failing because params was not a map object. So I assumed then that it was receiving a JSON array via the API because of PHP's funny dual-purpose (numeric and associative) array type being serialized to a JSON array.
